inc/ThirdParty/Hostings/WPEngine.php

```diff
@@ -0,0 +1,160 @@
+			'after_rocket_clean_domain'               => 'clean_wpengine',
+			'rocket_buffer'                           => [ 'add_footprint', 50 ],
+			'rocket_disable_htaccess'                 => 'disable_htaccess',
+		];
+	}
+
+	/**
+	 * Changes the text on the Varnish one-click block.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param array $settings Field settings data.
+	 *
+	 * @return array modified field settings data.
+	 */
+	public function varnish_addon_title( $settings ) {
+		$settings['varnish_auto_purge']['title'] = sprintf(
+			// Translators: %s = Hosting name.
+			__( 'Your site is hosted on %s, we have enabled Varnish auto-purge for compatibility.', 'rocket' ),
+			'WP Engine'
+		);
+
+		return $settings;
+	}
+
+	/**
+	 * Stop showing not valid notices with WP Engine.
+	 *
+	 * @since 3.6.1
+	 */
+	public function remove_notices() {
+		$container  = apply_filters( 'rocket_container', null );
+		$subscriber = $container->get( 'admin_cache_subscriber' );
+
+		remove_action( 'admin_notices', [ $subscriber, 'notice_advanced_cache_permissions' ] );
+	}
+
+	/**
+	 * Run WP Rocket preload bot after purged the Varnish cache via WP Engine Hosting.
+	 *
+	 * @since 3.6.1
+	 */
+	public function run_rocket_bot_after_wpengine() {
+		if ( ! wpe_param( 'purge-all' ) ) {
+			return;
+		}
+
+		if ( ! rocket_has_constant( 'PWP_NAME' ) ) {
+			return;
+		}
+
+		if ( ! check_admin_referer( rocket_get_constant( 'PWP_NAME' ) . '-config' ) ) {
+			return;
+		}
+
+		// Preload cache.
+		run_rocket_bot();
+		run_rocket_sitemap_preload();
+	}
+
+	/**
+	 * Call the cache server to purge the cache with WP Engine hosting.
+	 *
+	 * @since 3.6.1
+	 */
+	public function clean_wpengine() {
+		if ( method_exists( 'WpeCommon', 'purge_memcached' ) ) {
+			WpeCommon::purge_memcached();
+		}
+
+		if ( method_exists( 'WpeCommon', 'purge_varnish_cache' ) ) {
+			WpeCommon::purge_varnish_cache();
+		}
+	}
+
+	/**
+	 * Add WP Rocket footprint on Buffer.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param string $buffer HTML content.
+	 *
+	 * @return string HTML with WP Rocket footprint.
+	 */
+	public function add_footprint( $buffer ) {
+		if ( ! preg_match( '/<\/html>/i', $buffer ) ) {
+			return $buffer;
+		}
+
+		$footprint  = rocket_get_constant( 'WP_ROCKET_WHITE_LABEL_FOOTPRINT' )
+			? "\n" . '<!-- Optimized for great performance'
+			: "\n" . '<!-- This website is like a Rocket, isn\'t it? Performance optimized by ' . rocket_get_constant( 'WP_ROCKET_PLUGIN_NAME' ) . '. Learn more: https://wp-rocket.me';
+		$footprint .= ' -->';
+
+		return $buffer . $footprint;
+	}
+
+	/**
+	 * Disable dealing with htaccess if php version is 7.4 or above.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param bool $disable Optional. When true, disables.
+	 *
+	 * @return bool
+	 */
+	public function disable_htaccess( $disable = false ) {
+		// PHP version should be 7.4 or above.
+		if ( version_compare( PHP_VERSION, '7.4' ) >= 0 ) {
+			return true;
+		}
+
+		return (bool) $disable;
+	}
```

Comment:
@arunbasillal Is this the code you are referring to https://github.com/wp-media/wp-rocket/pull/2765#issuecomment-647566596? 

I do see a comment by @Tabrisrp https://github.com/wp-media/wp-rocket/issues/2345#issuecomment-642929298 which talks about before PHP 7.4 having a filter, which we have above. If the version is 7.4+, it's disabled; else, it's filtered. 

I'd assume the default should be `true` to disable, i.e. `$disable = true` for the filter.

@arunbasillal What is your understanding?

Reply:
@hellofromtonya The idea was never to check for PHP version as WP Engine already applies browser caching rules and compression (in fact Brotli, not Gzip) already even without WP Rocket. 

I tested this on a WP Engine website running PHP 7.3. So this entire check is pointless. Let's do it for EVERY WP Engine website as Remy suggested.

Reply:
@Tabrisrp @engahmeds3ed Shouldn't this filter default to `true` to disable on WPE?

Reply:
Removed the PHP version check and switched the param to `true` to disable by default https://github.com/wp-media/wp-rocket/pull/2765/commits/9d3dc869832eb0d79c1923ff9f3d6751c4acb651